src/cmd/linuxkit/run_hyperv.go

```diff
@@ -0,0 +1,264 @@
+		"-AutomaticStartAction", "Nothing",
+		"-AutomaticStopAction", "ShutDown",
+		"-CheckpointType", "Disabled",
+		"-MemoryStartupBytes", fmt.Sprintf("%dMB", *mem),
+		"-StaticMemory",
+		"-ProcessorCount", fmt.Sprintf("%d", *cpus))
+	if err != nil {
+		log.Fatalf("Failed to configure new VM: %v\n%s", err, out)
+	}
+
+	for i, d := range disks {
+		id := ""
+		if i != 0 {
+			id = strconv.Itoa(i)
+		}
+		if d.Size != 0 && d.Path == "" {
+			d.Path = *vmName + "-disk" + id + ".vhdx"
+		}
+		if d.Path == "" {
+			log.Fatalf("disk specified with no size or name")
+		}
+
+		if _, err := os.Stat(d.Path); err != nil {
+			if os.IsNotExist(err) {
+				log.Infof("Creating new disk %s %dMB", d.Path, d.Size)
+				_, out, err = poshCmd("New-VHD",
+					"-Path", fmt.Sprintf("'%s'", d.Path),
+					"-SizeBytes", fmt.Sprintf("%dMB", d.Size),
+					"-Dynamic")
+				if err != nil {
+					log.Fatalf("Failed to create VHD %s: %v\n%s", d.Path, err, out)
+				}
+			} else {
+				log.Fatalf("Problem accessing disk %s. %v", d.Path, err)
+			}
+		} else {
+			log.Infof("Using existing disk %s", d.Path)
+		}
+
+		_, out, err = poshCmd("Add-VMHardDiskDrive",
+			"-VMName", fmt.Sprintf("'%s'", *vmName),
+			"-Path", fmt.Sprintf("'%s'", d.Path))
+		if err != nil {
+			log.Fatalf("Failed to add VHD %s: %v\n%s", d.Path, err, out)
+		}
+	}
+
+	log.Info("Setting up boot from ISO")
+	_, out, err = poshCmd("Add-VMDvdDrive",
+		"-VMName", fmt.Sprintf("'%s'", *vmName),
+		"-Path", fmt.Sprintf("'%s'", isoPath))
+	if err != nil {
+		log.Fatalf("Failed add DVD: %v\n%s", err, out)
+	}
+	_, out, err = poshCmd(
+		fmt.Sprintf("$cdrom = Get-VMDvdDrive -vmname '%s';", *vmName),
+		"Set-VMFirmware", "-VMName", fmt.Sprintf("'%s'", *vmName),
+		"-EnableSecureBoot", "Off",
+		"-FirstBootDevice", "$cdrom")
+	if err != nil {
+		log.Fatalf("Failed set DVD as boot device: %v\n%s", err, out)
+	}
+
+	log.Info("Set up COM port")
+	_, out, err = poshCmd("Set-VMComPort",
+		"-VMName", fmt.Sprintf("'%s'", *vmName),
+		"-number", "1",
+		"-Path", fmt.Sprintf(`\\.\pipe\%s-com1`, *vmName))
+	if err != nil {
+		log.Fatalf("Failed set up COM port: %v\n%s", err, out)
+	}
+
+	log.Info("Start the VM")
+	_, out, err = poshCmd("Start-VM", "-Name", fmt.Sprintf("'%s'", *vmName))
+	if err != nil {
+		log.Fatalf("Failed start the VM: %v\n%s", err, out)
+	}
+
+	err = hypervStartConsole(*vmName)
+	if err != nil {
+		log.Infof("Console returned: %v\n", err)
+	}
+	hypervRestoreConsole()
+
+	if *keep {
+		return
+	}
+
+	log.Info("Stop the VM")
+	_, out, err = poshCmd("Stop-VM",
+		"-Name", fmt.Sprintf("'%s'", *vmName), "-Force")
+	if err != nil {
+		// Don't error out, could get an error if VM is already stopped
+		log.Infof("Stop-VM error: %v\n%s", err, out)
+	}
+
+	log.Info("Remove the VM")
+	_, out, err = poshCmd("Remove-VM",
+		"-Name", fmt.Sprintf("'%s'", *vmName), "-Force")
+	if err != nil {
+		log.Infof("Remove-VM error: %v\n%s", err, out)
+	}
+	// TODO(rn): Should we remove disks here?
```

Comment:
we leave them for the other local backends, so you can use them as persistent disks